File: Tests/CK.StObj.TypeScript.Tests/CrisLike/SampleCommands/NamedRecord.cs
namespace CK.StObj.TypeScript.Tests.CrisLike;

/// <summary>
/// Simple data record. Compatible with a IPoco field (no mutable reference).
/// </summary>
/// <param name="Value">The data value.</param>
/// <param name="Name">The data name.</param>
[TypeScript( SameFolderAs = typeof( ICommandAbs ) )]
public record struct NamedRecord( int Value, string Name );

File: Tests/CK.TypeScript.Tests/TSInlineTests/CrisLike_commands_and_results/ck-gen/Cmd/Abstract/NamedRecord.ts
import { CTSType } from '../../CK/Core/CTSType';

/**
 * Simple data record. Compatible with a IPoco field (no mutable reference).
 **/
export class NamedRecord {
public constructor(
/**
 * The data value.
 **/
public value: number = 0,
/**
 * The data name.
 **/
public name: string = "")
{
CTSType["CK.TypeScript.Tests.CrisLike.NamedRecord"].set( this );
}
readonly _brand!: {"88":any};
}
